handlers/practitioner_resource.go

```diff
@@ -216,10 +217,18 @@ func HandleGetPractitionerResource(svc dao.Service) http.Handler {
 			return
 		}
 
-		practitionerResourceDaos = append(practitionerResources, practitionerResourceDaos...)
+		// check if each practitioner has valid transactionID
+		for _, practitionerResource := range practitionerResources {
```

Comment:
Seeing 'range' used here makes me wonder whether there should ever be more than one result for a specific practitioner ID, & whether we should return an error if there is? (same question for line 402 in the HandleGetPractitionerAppointment function).

Reply:
That function has the capability of accepting array of practitioners hence would return array even for a single supplied practitionerID. The function is well position now and for future. In addition aggregation always returns arrays and we are using aggregation here and must be encouraged going forward. To answer your question directly, we do not need error handling because of the stated reasons. Thanks

Reply:
I understand that the practitionerResources are returned in the form of an array, my question is whether in this particular scenario (where we have passed in a single id) we should be checking that a single result has been returned rather than iterating over the range and returning the last match found. (I.e. check that len(practitionerResources) is 1 -> check txn id against that single result if it is, return an error if not).

Reply:
Resolving after discussion as this is a low-priority question & we are in agreement that in practice there would only ever be one.